src/services/gateways/aws_2.ts

```diff
@@ -1868,4 +1868,30 @@ export class AWS {
     });
     return res.RouteTables;
   }
+
+  async createInstanceProfile(roleName: string) {
+    await this.iamClient.createInstanceProfile({
+      InstanceProfileName: roleName,
+    });
+  }
+
+  async attachRoleToInstanceProfile(roleName: string) {
+    await this.iamClient.addRoleToInstanceProfile({
+      InstanceProfileName: roleName,
+      RoleName: roleName,
+    });
+  }
+
+  async deleteInstanceProfile(roleName: string) {
+    await this.iamClient.deleteInstanceProfile({
+      InstanceProfileName: roleName,
+    });
+  }
+
+  async detachRoleToInstanceProfile(roleName: string) {
+    await this.iamClient.removeRoleFromInstanceProfile({
+      InstanceProfileName: roleName,
+      RoleName: roleName,
+    });
+  }
```

Comment:
Can you convert these to the macro form?

Reply:
I thought about it but I wasn't sure if do it in this PR or keep that change in a separate PR